src/libraries/System.Private.CoreLib/src/System/Numerics/Quaternion.cs

```diff
@@ -188,31 +188,49 @@ public static Quaternion Identity
         /// <remarks>The <see cref="Quaternion.op_Multiply" /> method defines the operation of the multiplication operator for <see cref="Quaternion" /> objects.</remarks>
         public static Quaternion operator *(Quaternion value1, Quaternion value2)
         {
-            Quaternion ans;
-
-            float q1x = value1.X;
-            float q1y = value1.Y;
-            float q1z = value1.Z;
-            float q1w = value1.W;
-
-            float q2x = value2.X;
-            float q2y = value2.Y;
-            float q2z = value2.Z;
-            float q2w = value2.W;
-
-            // cross(av, bv)
-            float cx = q1y * q2z - q1z * q2y;
-            float cy = q1z * q2x - q1x * q2z;
-            float cz = q1x * q2y - q1y * q2x;
-
-            float dot = q1x * q2x + q1y * q2y + q1z * q2z;
-
-            ans.X = q1x * q2w + q2x * q1w + cx;
-            ans.Y = q1y * q2w + q2y * q1w + cy;
-            ans.Z = q1z * q2w + q2z * q1w + cz;
-            ans.W = q1w * q2w - dot;
-
-            return ans;
+            if (Vector.IsHardwareAccelerated)
+            {
+                // Concatenate rotation is actually q2 * q1 instead of q1 * q2.
+                // So that's why value2 goes q1 and value1 goes q2.
+                float q1w = value2.W;
+                float q2w = value1.W;
+
+                Vector3 q1V = new Vector3(value2.X, value2.Y, value2.Z);
+                Vector3 q2V = new Vector3(value1.X, value1.Y, value1.Z);
+
+                return new Quaternion(
+                    q1w * q2V + q2w * q1V + Vector3.Cross(q1V, q2V),
+                    q1w * q2w - Vector3.Dot(q1V, q2V)
+                );
+            }
```

Comment:
I would expect something like the following to perform better:
```csharp
if (Vector128.IsHardwareAccelerated)
{
    var left = value1.AsVector128();
    var right = value2.AsVector128();

    var result = right * left.GetElement(3);
    result += (Vector128.Shuffle(right, Vector128.Create(3, 2, 1, 0)) * left.GetElement(0)) * Vector128.Create(+1.0f, -1.0f, +1.0f, -1.0f);
    result += (Vector128.Shuffle(right, Vector128.Create(2, 3, 0, 1)) * left.GetElement(1)) * Vector128.Create(+1.0f, +1.0f, -1.0f, -1.0f);
    result += (Vector128.Shuffle(right, Vector128.Create(1, 0, 3, 2)) * left.GetElement(2)) * Vector128.Create(-1.0f, +1.0f, +1.0f, -1.0f);
    return result.AsQuaternion();
}
```

Reply:
Thanks, that is indeed quite a lot faster, and the results still looked correct.

```
| Method                 | Mean      | Error     | StdDev    |
|----------------------- |----------:|----------:|----------:|
| MulOperator            | 4.5998 ns | 0.0843 ns | 0.0788 ns |
| VectorMul              | 2.9167 ns | 0.0145 ns | 0.0128 ns |
| IntrinsicMulGetElement | 0.8482 ns | 0.0113 ns | 0.0106 ns |
```

I've modified it to use the GetElementUnsafe method, i saw other Numerics types use this too.